src/fighter_state.rs

```diff
@@ -0,0 +1,747 @@
+
+        // Set our figher velocity to the knock back velocity
+        **velocity = knocked_back.velocity;
+    }
+}
+
+/// Update dying players
+fn dying(
+    mut commands: Commands,
+    mut fighters: Query<(Entity, &mut Animation, &mut LinearVelocity), With<Dying>>,
+) {
+    for (entity, mut animation, mut velocity) in &mut fighters {
+        // Start playing the dying animation if it isn't already
+        if animation.current_animation.as_deref() != Some(Dying::ANIMATION) {
+            **velocity = Vec2::ZERO;
+            animation.play(Dying::ANIMATION, false);
+
+        // When the animatino is finished, despawn the fighter
+        } else if animation.is_finished() {
+            commands.entity(entity).despawn_recursive();
+        }
+    }
+}
+
+/// Throw the item in the player's inventory
+fn throwing(
+    mut commands: Commands,
+    mut fighters: Query<
+        (
+            Entity,
+            &Transform,
+            &Facing,
+            &Stats,
+            &mut Inventory,
+            &mut Health,
+        ),
+        With<Throwing>,
+    >,
+    item_assets: Res<Assets<ItemMeta>>,
+) {
+    for (entity, fighter_transform, facing, stats, mut inventory, mut health) in &mut fighters {
+        // If the player has an item in their inventory
+        if let Some(meta_handle) = inventory.take() {
+            // If the item asset has loaded
+            if let Some(item) = item_assets.get(&meta_handle) {
+                // Check what kind of item this is.
+                //
+                // TODO: We should probably create a flexible item system abstraction similar to the
+                // fighter state abstraction so that items can flexibly defined without a
+                // centralized enum.
+                match item.kind {
+                    ItemKind::Throwable { .. } => {
+                        // Throw the item!
+                        commands.spawn_bundle(Projectile::from_thrown_item(
+                            fighter_transform.translation + consts::THROW_ITEM_OFFSET.extend(0.0),
+                            item,
+                            facing,
+                        ));
+                    }
+                    ItemKind::Health {
+                        health: item_health,
+                    } => {
+                        // Refill player's health
+                        **health = (**health + item_health).clamp(0, stats.max_health);
+                    }
+                }
+
+            // If the item asset isn't loaded yet
+            } else {
+                // This shouldn't happen because we make sure our assets are loaded before we start
+                // the game. But just in case, we'll want to know about it if it does happen.
+                warn!("Thrown item not spawned because it has not been loaded yet");
+            }
+        }
+
+        // Throwing is an "instant" state, that is removed at the end of every frame
```

Comment:
Will probably want this to have an animation and not be instant in future.

Reply:
Good point.

Cool thing is, that's super easy to change!

We just have to make the throwing state transition non-additive so that it takes over for the other states, and then we can be free to play an animation during the `throwing` system.